src/pair.ts

```diff
@@ -31,12 +31,20 @@ export abstract class Pair {
 		swappaPairAddress: Address,
 	}>;
 	public abstract refresh(): Promise<void>;
-	public swapData(inputToken: Address): SwapData {
+	public swapData(inputToken: Address, inputAmount?: BigNumber): SwapData {
 		return {
 			addr: this.swappaPairAddress,
-			extra: this.swapExtraData(inputToken),
+			extra: this.swapExtraDataWithInputAmount(inputToken, inputAmount),
 		}
 	}
+	protected swapExtraDataWithInputAmount(inputToken: Address, inputAmount?: BigNumber): string {
+		const swapExtraData = this.swapExtraData(inputToken)
+		if (!inputAmount) {
+			return swapExtraData
+		}
+		const inputAmountData = inputAmount.integerValue().toString(16).padStart(64, "0")
```

Comment:
take integer value, convert to hex, pad to 64 hex characters = 256 binary bits
append at end of swap extra data